src/lightning/data/streaming/data_processor.py

```diff
@@ -101,6 +101,16 @@ def _wait_for_file_to_exist(s3: S3Client, obj: parse.ParseResult, sleep_time: in
                 raise e
 
 
+def _wait_for_disk_usage_higher_than_threshold(threshold_in_gb: int = 25, sleep_time: int = 3) -> None:
+    usage = shutil.disk_usage("/")
+
+    while (usage.free / 1000 / 1000 / 1000) <= threshold_in_gb:
+        sleep(sleep_time)
+        usage = shutil.disk_usage("/")
```

Comment:
should probably check the disk usage with the path where we are writing to. E.g. instead of `/` it should be the "cache dir" probably.

Reply:
I think it is safer to check the overall machine state to avoid any failures.

Reply:
My argument is that it would be a costly mistake not to measure the disk usage where we are saving. For example, if your /home is not on the same disk as /. The function could simply take a path as input, like

```py
 _wait_for_disk_usage_higher_than_threshold(input_dir.path)
```

Reply:
The `root` should include everything on the machine, so it would independent on wherever is home.